File: android/src/main/kotlin/ru/kardenvan/flutter_yandex_ads_plus/ads/AdParameters.kt
package ru.kardenvan.flutter_yandex_ads_plus.ads

import android.location.Location
import com.yandex.mobile.ads.common.AdTheme


data class AdParameters(
    val age: String?,
    val biddingData: String?,
    val contextQuery: String?,
    val contextTags: List<String>?,
    val gender: String?,
    val preferredTheme: AdTheme?,
    val location: Location?,
    val custom: Map<String, String>?,
)

File: lib/core/ad_parameters/ad_parameters.dart
import 'dart:ui';

import 'location.dart';

export 'location.dart';

/// Data class containing additional ad request parameters.
///
class AdParameters {
  const AdParameters({
    this.age,
    this.biddingData,
    this.contextQuery,
    this.contextTags,
    this.gender,
    this.custom,
    this.brightness,
    this.location,
  });

  final String? age;
  final String? biddingData;
  final String? contextQuery;
  final List<String>? contextTags;
  final String? gender;
  final Map<String, String>? custom;
  final Brightness? brightness;
  final Location? location;
}
